### PR TITLE
Reuse compliance

### PR DESCRIPTION
**Pull request by Piccions**
This PR improves the licensing scheme of the project files by adopting REUSE guidelines (https://reuse.software/).

### CODE REVIEW

File: bump-version.sh [hunk 1]
```diff
@@ -1,43 +0,0 @@
-#!/bin/bash
```
Comment (alessandropellegrini) [inline bump-version.sh:1]: If we switch to a python script, also the instructions in this file should be updated accordingly:
https://github.com/ROOT-Sim/core/blob/f6618296aadbeef5eb9230a8dc9fcf7490838d27/docs/CONTRIBUTING.md

File: .reuse/dep5 [hunk 2]
```diff
@@ -0,0 +1,33 @@
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: ROOT-Sim
+Upstream-Contact: Andrea Piccione <piccione@diag.uniroma1.it>
```
Comment (alessandropellegrini) [inline .reuse/dep5:3]: Change to: HPDCS Research Group <rootsim@googlegroups.com>

File: src/ROOT-Sim.h [hunk 3]
```diff
@@ -10,23 +10,8 @@
  * a simulation model. All function prototypes exposed to the application
  * developer are exposed and defined here.
  *
- * @copyright
```
Comment (alessandropellegrini) [inline src/ROOT-Sim.h:13]: The @copyright tag allows to generate the Doxygen documentation consistently. Should we keep it as is and have SPDX in addition? Should we keep it but have only SPDX? Should we drop this Doxygen tag?

File: src/arch/thread.c [hunk 4]
```diff
@@ -11,23 +11,8 @@
  * once all threads are synchronized on the barrier, the function returns
  * true to only one of them.
  *
- * @copyright
```
Comment (alessandropellegrini) [inline src/arch/thread.c:14]: The @copyright tag allows to generate the Doxygen documentation consistently. Should we keep it as is and have SPDX in addition? Should we keep it but have only SPDX? Should we drop this Doxygen tag?
This should apply to all changes in this PR :cry:

File: src/core/core.h [hunk 5]
```diff
@@ -54,6 +39,8 @@ __extension__({				\
 })
 
 #ifndef CACHE_LINE_SIZE // TODO: calculate and inject at build time
+/// The size of a cpu cache line
```
Comment (alessandropellegrini) [inline src/core/core.h:42]: Maybe this is breaking Doxygen syntax and only the second comment is being included in the doc.

File: src/core/init.c [hunk 6]
```diff
@@ -211,7 +211,7 @@ struct ap_section ap_sects[] = {
 struct ap_settings ap_sets = {
 	"ROOT-Sim",	// TODO properly fill these fields
 	"Proper version string",
```
Comment (alessandropellegrini) [inline src/core/init.c:213]: This should be populated with the version in meson build files.

File: src/log/log.h [hunk 7]
```diff
@@ -1,51 +1,54 @@
...
+ * SPDX-FileCopyrightText: 2008-2021 HPDCS Group <rootsim@googlegroups.com>
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
 #pragma once
 
 #include <stdbool.h>
 
 #ifndef LOG_LEVEL
+/// The minimum logging level supported at compile time
+/** Compiler optimizations remove log calls with lower level than this one */
```
Comment (alessandropellegrini) [inline src/log/log.h:17]: Maybe this comment breaks Doxygen syntax and only the last one is considered for the documentation.

File: test/test/fail_init.c [hunk 8]
```diff
@@ -1,10 +1,25 @@
+/**
+ * @file test/test/fail_init.c
+ *
+ * @brief Test: expected failure, \ref test_config.test_init_fnc failure
+ *
+ * @test Tests that a failure in the test initialization function
+ * \ref test_config.test_init_fnc call causes a test failure.
```
Comment (alessandropellegrini) [inline test/test/fail_init.c:7]: @ref?

**Review by alessandropellegrini — CHANGES_REQUESTED** [review]
Good job! Thanks! Some minor issues in the comment.
One of these minor issues could be painful to address :cry:

File: src/ROOT-Sim.h (hunk 3 above)
Comment (Piccions) [reply src/ROOT-Sim.h:13]: This problem is addressed by a proper doxygen filter script, which you can find at _scripts/doxy_filter.py_
SPDX tags are transformed into equivalent **@copyright** Doxygen tags.

File: .reuse/dep5 (hunk 2 above)
Comment (Piccions) [reply .reuse/dep5:3]: Yep, I missed that! Anyway I'll change it to "HPDCS Group <rootsim@googlegroups.com>" for consistency with the other SPDX headers, if we agree.

File: src/core/init.c (hunk 6 above)
Comment (Piccions) [reply src/core/init.c:213]: Good call! This will be fixed in the next commit.

File: src/log/log.h (hunk 7 above)
Comment (Piccions) [reply src/log/log.h:17]: Actually this syntax is suggested by the Doxygen documentation:

- the first line, starting with **///**, is recognized as the brief description,
- the second line, enclosed in **/**** and ***/**, is mapped to the "long" description.

This syntax works only if both sections are one line long.

File: src/core/core.h (hunk 5 above)
Comment (Piccions) [reply src/core/core.h:42]: I answered to your other similar remark

File: test/test/fail_init.c (hunk 8 above)
Comment (Piccions) [reply test/test/fail_init.c:7]: I'll change those to the more syntactically consistent _@ref_ tags.

File: bump-version.sh (hunk 1 above)
Comment (Piccions) [reply bump-version.sh:1]: This will be fixed in the next commit!

**Review by alessandropellegrini — APPROVED** [review]

**Pull request merged**